sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -270,6 +271,18 @@ private[sql] object PartitioningUtils {
   private val upCastingOrder: Seq[DataType] =
     Seq(NullType, IntegerType, LongType, FloatType, DoubleType, StringType)
 
+  def checkPartitionColumnOfValidDataType(
+      schema: StructType,
+      partitionColumns: Array[String]): Unit = {
+
+    ResolvedDataSource.partitionColumnsSchema(schema, partitionColumns).foreach { field =>
+      field.dataType match {
+        case _: AtomicType | NullType => // OK
```

Comment:
I have `NullType` here because I think it's possible the schema was inferred from existing partition columns . For example, `DataFrameWriter.insert` into a `HadoopFsRelation`, in this case, it's possible that the column type can be `NullType`. Is this possible?